tests/src/test_sprite.cpp

```
@@ -0,0 +1,277 @@
+// ---------------------------------------------------------------------------
+// Test helpers
+// ---------------------------------------------------------------------------
+
+namespace {
+    // Minimal IImage stub — no GPU resources, satisfies SpriteSheet's non-null invariant.
+    class DummyImage : public IImage {
+    public:
+        int GetWidth() const override { return 64; }
+        int GetHeight() const override { return 64; }
+        std::shared_ptr<ITexture> GetTexture() const override { return nullptr; }
+        void ImGui(IntVec2 const&, FloatVec2 const&, FloatVec2 const&) const override {}
```

Clang (14.0.6)

[error] all parameters should be named in a function

(readability-named-parameter,-warnings-as-errors)